restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java

```diff
@@ -1564,6 +1575,23 @@ public void execute(final Object message) throws Exception {
                 inputType = Collect.Type.parseOrDefault(typeAttr.value(), Collect.Type.DTMF);
             }
 
+            if (inputType.equals(Collect.Type.SPEECH) || inputType.equals(Collect.Type.DTMF_SPEECH)) {
+                ExtensionController ec = ExtensionController.getInstance();
+                final IExtensionFeatureAccessRequest far = new FeatureAccessRequest(FeatureAccessRequest.Feature.ASR, accountId);
+                ExtensionResponse er = ec.executePreInboundAction(far, extensions);
+                if (!er.isAllowed()) {
+                    if (logger.isDebugEnabled()) {
+                        final String errMsg = "ASR feature is not allowed";
+                        logger.debug(errMsg);
+                    }
+                    String errMsg = "ASR feature is not allowed";
+                    notification(WARNING_NOTIFICATION, 11001, errMsg);
+                    call.tell(new Hangup(errMsg), source);
+                    return;
+                }
+                ec.executePostOutboundAction(er, extensions);
```

Comment:
should we delay the invocation of PostAction to when the ASR is actually executed?

Reply:
@jaimecasero, in general, the `post` action should be executed after the `pre` action. 

At the point of your comment,  the call is already answered and maybe had already executed some RCML verbs, the interpreter started executions of the `gather` verb with `asr` which account was not permitted and thus interpreter hangup the call. I think this is the proper place for `post` action method.